### ISSUE TITLE
[FEATURE] UI customization options

### ISSUE THREAD

**Issue by aaronleopold** [stumpapp/stump#245]
**Is your feature request related to a problem? Please describe.**

A suite of options for customizing the UI on a per-user basis (i.e. manage my own preferences for what is displayed)

**Describe the solution you'd like**

The following are things I personally want for my own hosted server, if anyone else has any ideas/suggestions let me know. Otherwise, I'm satisfying my own needs only 😛 

- [ ] Rearrange order sidebar/topbar items
- [ ] Hide sidebar/topbar items
    - [ ] Include the ability to hide the shortcuts, e.g. create new library button
- [ ] Rearrange home page lists (e.g. recently read, recently added, etc.) and add additional content
    - [ ] I'd also like to add more here, this is one of the more dated parts of Stump and there are a lot more options available. I also think varying the type of content (i.e. list vs grid group vs table, etc) would be really visually good
- [ ] Support a paged, table layout for the explore books scene
    - [ ] After working with Stump's smart lists, I really enjoyed using a table for browsing my stuff. This is a large effort and will likely be separated to its own ticket, though.
- [ ] Compact display mode to reduce overall spacing
- [ ] Prefer colors vs prefer monochrome

**Comment by devy1540** [comment]
I'd like to help with the home page customization mentioned here. In particular, I'd like to be able to hide sections such as “Recently added books” and “Recently added series,” and change the order of the sections I keep.

Would a first PR covering the following be welcome?

- Show or hide individual home page sections.
- Reorder the existing sections.
- Save these preferences per user, keeping the current layout as the default.

If this would be useful, I'd be happy to work on the implementation. Is anyone already working on this, or is there a preferred approach I should follow before starting?

**Comment by aaronleopold** [comment]
> Would a first PR covering the following be welcome?
> 
> - Show or hide individual home page sections.
> - Reorder the existing sections.
> - Save these preferences per user, keeping the current layout as the default

Yes and nobody else is working on it to my knowledge. There are already some [models relevant to this](https://github.com/stumpapp/stump/blob/nightly/crates/models/src/shared/arrangement.rs), so you have something to work off of.


> is there a preferred approach I should follow before starting?

One thing I'll say is in the arrangements I had soft plans to remove that "locked" status, so if you can avoid that in the mutations that'd be preferred. I would also eventually want to hook this up to the mobile app, as well
